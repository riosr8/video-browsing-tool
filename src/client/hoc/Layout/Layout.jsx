import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Aux from '../Auxdir/Auxfile';
import NavBar from '../../components/NavBar/NavBar';

const Layout = props => {
  const { children, isAuthenticated, userId, activePage, sortKey, sortOrder } = props;
  return (
    <Aux>
      <NavBar
        isAuthenticated={isAuthenticated}
        userId={userId}
        activePage={activePage}
        sortKey={sortKey}
        sortOrder={sortOrder}
      />
      <main>{children}</main>
    </Aux>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

const mapStateToProps = state => {
  return {
    isAuthenticated: state.auth.token !== null,
    userId: state.auth.userId,
    activePage: state.dashboard.activePage,
    sortKey: state.dashboard.sortKey,
    sortOrder: state.dashboard.sortOrder,
  };
};

export default connect(mapStateToProps)(Layout);

Layout.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  userId: PropTypes.string,
  activePage: PropTypes.number.isRequired,
  sortKey: PropTypes.string.isRequired,
  sortOrder: PropTypes.string.isRequired,
};
