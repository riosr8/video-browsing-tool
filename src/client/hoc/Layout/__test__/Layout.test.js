import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import Layout from '../Layout';

const tests = [
  {
    testName: 'should render without crashing if token present',

    fakeStore: {
      default: () => {},
      subscribe: () => {},
      dispatch: () => {},
      getState: () => ({
        auth: { token: 'token' },
        dashboard: { activePage: 1, sortKey: 'time', sortOrder: 'asc' },
      }),
    },
  },
  {
    testName: 'should render without crashing if no token present',

    fakeStore: {
      default: () => {},
      subscribe: () => {},
      dispatch: () => {},
      getState: () => ({
        auth: { token: null },
        dashboard: { activePage: 1, sortKey: 'time', sortOrder: 'asc' },
      }),
    },
  },
];

describe('Layout HOC', () => {
  tests.forEach(test => {
    it(test.testName, () => {
      const tree = renderer
        .create(
          <Provider store={test.fakeStore}>
            <MemoryRouter>
              <Layout>
                <div>Test</div>
              </Layout>
            </MemoryRouter>
          </Provider>,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
