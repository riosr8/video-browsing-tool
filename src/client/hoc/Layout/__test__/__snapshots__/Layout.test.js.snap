// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout HOC should render without crashing if no token present 1`] = `
Array [
  <div>
    <div
      className="ui huge inverted top attached menu"
      style={
        Object {
          "backgroundColor": "#e11b37",
        }
      }
    >
      <div
        className="ui container"
      >
        <div
          className="header item"
          onClick={[Function]}
        >
          Behaviorics
        </div>
        <a
          aria-current="page"
          className="item active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <div
          className="right menu"
        >
          <a
            aria-current={null}
            className="item"
            href="/signup"
            onClick={[Function]}
          >
            Sign Up
          </a>
          <a
            aria-current={null}
            className="item"
            href="/login"
            onClick={[Function]}
          >
            Log In
          </a>
        </div>
      </div>
    </div>
  </div>,
  <main>
    <div>
      Test
    </div>
  </main>,
]
`;

exports[`Layout HOC should render without crashing if token present 1`] = `
Array [
  <div>
    <div
      className="ui huge inverted top attached menu"
      style={
        Object {
          "backgroundColor": "#e11b37",
        }
      }
    >
      <div
        className="ui container"
      >
        <div
          className="header item"
          onClick={[Function]}
        >
          Behaviorics
        </div>
        <a
          aria-current="page"
          className="item active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="item"
          href="/dashboard?pageNo=1&sortVal=time&order=asc"
          onClick={[Function]}
        >
          Dashboard
        </a>
        <a
          aria-current={null}
          className="item"
          href="/process-status"
          onClick={[Function]}
        >
          Process Status
        </a>
        <div
          className="right menu"
        >
          <div
            className="item"
            onClick={[Function]}
          >
            Signed in as undefined
          </div>
          <a
            aria-current={null}
            className="item"
            href="/logout"
            onClick={[Function]}
          >
            Log Out
          </a>
        </div>
      </div>
    </div>
  </div>,
  <main>
    <div>
      Test
    </div>
  </main>,
]
`;
